_transcripts/033-rust-1.50-1.51.md

```diff
@@ -0,0 +1,1232 @@
+for more constrained, or— not constrained, but more involved use cases. So one
+is, you currently can only use, I think it's bools, characters and integer
+types. You can only be const generic over those kinds of types. And I think,
+sort of, in the ultimate version, right, you'll be able to be generic over any
+constant value of any type, or of any particular type you give. That's not
+currently something you can do. You also currently can't have the constant
+generic have a value that depends on other generic parameters. So you can't say
+that, like, I'm generic over some `const M`. And this other type is going to be
+generic over `M + 1`, for example. Like, you can't put in expressions that are
+themselves generic.
+
+__Ben__: And I think expressions in general are kind of limited right now. You
+can't have, like a `where` clause for const generics that are like, you know,
+`where X = M + N` or anything.
+
+__Jon__: Yeah, exactly. And it also can't depend on like, other generic types.
+So if you have some type that's generic over `T`, you can't use const generic
+inside of there, and then pass in, like, `mem::size_of::<T>()`. That's also not
+supported yet. So there are some limitations, but they're ones that I think are
+expected to be lifted over time. And it was more like the current set that we
+have enables so many things that weren't possible before. So we should get that
+out the door, because it's just super valuable in and of itself. And then over
+time we can relax the restrictions that are in place.
+
+__Ben__: Yeah, there's a great blog post on the Rust Insider blog that we can
+link to. Well, let's assume that we have [linked to it][episode] and you can
+click it below.
+
+__Jon__: That future selves have already have had added links to.
+
+__Ben__: That, one, future yourself is now remembering to link, and that future
+listeners are now remembering to actually click on.
+
+__Jon__: Nice. The the other thing that's really cool, I think, and that that's
+sort of related to this const-generic-ification, is that now we finally have a
+way to take an array and turn it into an owned iterator. With a slight caveat.
+But do you want to talk a little bit about this?
+
+__Ben__: Yeah, sure. So, just briefly, `for` loops in Rust: pretty cool. You can
+kind of, like, if you have, say, a vector, you can just say
+`for thing in vector` and then curly brace and it gives you an array. And that
+is a pretty nice little piece of syntax that any user type can opt into, not
+just the standard types, by implementing the `IntoIter` trait. Actually, it's
+specifically the `iter::IntoIter` trait. *(editor's note: the trait is called
+`IntoIterator`, which contains a method called `into_iter`)* And so if you do
+this, then whenever you use this type in a `for` loop, it will implicitly,
+behind the scenes, call `into_iter` on that type, and then iterate over that
+value that it gives you.
```

Comment:
37:43 Is this editor's note OK? I try to avoid fact-checking, but this is a bit confusing.

Reply:
I think 37:43 isn't the right time code? I have it at 30:43. Regardless, yes, I think the note is fine. There's also the `IntoIter` associated type, but I don't think it's important, and your clarification is a good one.